### PR TITLE
planner: update Plan Cache doc

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by songrijie — APPROVED** [review]
rest LGTM

**Review by AilinKid — APPROVED** [review]

File: choose-index.md [hunk 1]
```diff
@@ -403,4 +403,74 @@ mysql> EXPLAIN SELECT /*+ use_index_merge(t3, idx) */ * FROM t3 WHERE ((1 member
 |   └─TableFullScan_6     | 10000.00 | cop[tikv] | table:t3      | keep order:false, stats:pseudo                                                                                                                                                                               |
 +-------------------------+----------+-----------+---------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 3 rows in set, 2 warnings (0.00 sec)
+```
+
+### 多值索引与 Plan Cache
+
+如果使用多值索引的查询包含 `json_contains` 或 `json_overlaps` 且他们会直接影响索引选择，则计划无法被缓存。如果不存在此问题，则可以缓存。
+
```
Comment (hfxsd) [inline choose-index.md:411]:
```suggestion

下面是可以缓存的例子：

```

File: choose-index.md [hunk 2]
```diff
@@ -403,4 +403,76 @@ mysql> EXPLAIN SELECT /*+ use_index_merge(t3, idx) */ * FROM t3 WHERE ((1 member
 |   └─TableFullScan_6     | 10000.00 | cop[tikv] | table:t3      | keep order:false, stats:pseudo                                                                                                                                                                               |
 +-------------------------+----------+-----------+---------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 3 rows in set, 2 warnings (0.00 sec)
+```
+
+### 多值索引与 Plan Cache
```
Comment (Oreoxmt) [inline choose-index.md:408]:
```suggestion
### 多值索引与执行计划缓存
```

File: choose-index.md [hunk 3]
```diff
@@ -403,4 +403,76 @@ mysql> EXPLAIN SELECT /*+ use_index_merge(t3, idx) */ * FROM t3 WHERE ((1 member
...
+
+通过 `member of` 条件选择到多值索引的执行计划可以被缓存；如果执行计划是通过 `json_contains` 或 `json_overlaps` 函数选择到多值索引，则该计划暂时无法被缓存。
```
Comment (Oreoxmt) [inline choose-index.md:410]:
```suggestion
通过 `member of` 条件选择到多值索引的执行计划可以被缓存。如果执行计划是通过 `json_contains` 或 `json_overlaps` 函数选择到多值索引，则该计划暂时无法被缓存。
```

**Review by Oreoxmt — APPROVED** [review]

File: choose-index.md [hunk 4]
```diff
@@ -403,4 +403,76 @@ mysql> EXPLAIN SELECT /*+ use_index_merge(t3, idx) */ * FROM t3 WHERE ((1 member
 |   └─TableFullScan_6     | 10000.00 | cop[tikv] | table:t3      | keep order:false, stats:pseudo                                                                                                                                                                               |
 +-------------------------+----------+-----------+---------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 3 rows in set, 2 warnings (0.00 sec)
+```
+
+### 多值索引与执行计划缓存
+
+通过 `member of` 条件选择的多值索引的执行计划可以被缓存。如果执行计划是通过 `json_contains` 或 `json_overlaps` 函数选择的多值索引，则该计划暂时无法被缓存。
```
Comment (hfxsd) [inline choose-index.md:410]:
```suggestion
通过 `member of` 条件选择的多值索引的执行计划可以被缓存。如果执行计划是通过 `JSON_CONTAINS()` 或 `JSON_OVERLAPS()` 函数选择的多值索引，则该计划暂时无法被缓存。
```

**Pull request merged**